llvm_objects.py

```diff
@@ -0,0 +1,238 @@
+        while l_block:
+            current_block = W_Block(l_block, w_module)
+            if prev:
+                prev.w_next_block = current_block
+            prev = current_block
+            self._set_block(l_block, current_block)
+            l_block = LLVMGetNextBasicBlock(l_block)
+
+    def get_block(self, l_block):
+        ''' Returns the W_Block representation of a given block of
+            this function.'''
+
+        if self.has_key(l_block):
+            addr = rffi.cast(rffi.INT, l_block)
+            offset = self._get_block_off(addr)
+            return self.w_blocks[offset]
+        else:
+            raise LookupError(rffi.charp2str(LLVMPrintValueToString(l_block)))
+
+    @jit.elidable_promote()
+    def _get_block_off(self, l_block):
+        offset = self.block_offsets.get(l_block, -1)
+        if offset == -1:
+            offset = len(self.w_blocks)
+            self.block_offsets[rffi.cast(rffi.INT, l_block)] = offset
+            self.w_blocks.append(None)
+        return offset
+
+    def _set_block(self, l_block, w_block):
+        addr = rffi.cast(rffi.INT, l_block)
+        offset = self._get_block_off(addr)
+        self.w_blocks[offset] = w_block
+
+    def has_key(self, l_block):
+        ''' Checks if this W_Function contains a given block. '''
+
+        addr = rffi.cast(rffi.INT, l_block)
+        offset = self.block_offsets.get(addr, -1)
+        if offset == -1:
+            return False
+        return True
+
+    def get_first_block(self):
+        return self.w_blocks[0]
+
+class W_Block(object):
+    ''' Represents a wrapper for an LLVMBasicBlockRef. '''
+
+    def __init__(self, l_block, w_module):
+        self.w_instructions = self.get_instructions(l_block, w_module)
+        self.w_next_block = None
+        self.l_value = LLVMValueAsBasicBlock(l_block)
+
+    def get_instructions(self, l_block, w_module):
+        l_instr = LLVMGetFirstInstruction(l_block)
+        w_instrs = []
+        prev = None
+        while l_instr:
+            current_instr = W_Instruction(l_instr, w_module)
+            if prev:
+                prev.w_next_instr = current_instr
+            prev = current_instr
+            w_instrs.append(current_instr)
+            l_instr = LLVMGetNextInstruction(l_instr)
+        return w_instrs
+
+    def get_first_instruction(self):
+        return self.w_instructions[0]
+
+    def __eq__(self, other):
+        if isinstance(other, W_Block):
+            return self.w_instructions == other.w_instructions and\
+                   self.l_value == other.l_value
+        return NotImplemented
+
+class W_Instruction(object):
+    ''' Represents a wrapper for an LLVMValueRef that is an instruction. '''
+
+    def __init__(self, l_instr, w_module):
+        self.addr = rffi.cast(rffi.INT, l_instr)
+        self.l_operands = self.get_operands(l_instr, w_module)
+        self.opcode = LLVMGetInstructionOpcode(l_instr)
+        self.w_next_instr = None
+        if self.opcode == LLVMBr:
```

Comment:
Seems a bit unnecessary since right now I only have a single type of instruction that could be conditional. However, I'll still do it your way because I might add some other conditional instructions later on.